front/lib/resources/group_resource.ts

```diff
@@ -366,70 +368,207 @@ export class GroupResource extends BaseResource<GroupModel> {
     return groups.map((group) => new this(GroupModel, group.get()));
   }
 
-  async addMember(
+  async getActiveMembers(auth: Authenticator): Promise<UserResource[]> {
+    const owner = auth.getNonNullableWorkspace();
+    if (!this.activeMembers) {
+      const memberships = await GroupMembershipModel.findAll({
+        where: {
+          groupId: this.id,
+          workspaceId: owner.id,
+          startAt: { [Op.lte]: new Date() },
+          [Op.or]: [{ endAt: null }, { endAt: { [Op.gt]: new Date() } }],
+        },
+      });
+
+      this.activeMembers = await UserResource.listByModelIds(
```

Comment:
I'm not a big fan of doing this in methods, either it should live in the constructor or it should not store anything in class. By doing so here data could be staled or it requires extreme hygiene to make sure that we always think about re-setting it properly.

Reply:
removed